tests/integration/test_quantiles.py

```diff
@@ -0,0 +1,434 @@
+    )
+    # print(np_q_out)
+
+    assert q_out.shape == np_q_out.shape
+    assert q_out.dtype == np_q_out.dtype
+
+    assert allclose(np_q_out, q_out, atol=eps)
+
+
+@pytest.mark.parametrize(
+    "str_method",
+    (
+        "inverted_cdf",
+        "averaged_inverted_cdf",
+        "closest_observation",
+        "interpolated_inverted_cdf",
+        "hazen",
+        "weibull",
+        "linear",
+        "median_unbiased",
+        "normal_unbiased",
+        "lower",
+        "higher",
+        "midpoint",
+        "nearest",
+    ),
+)
+@pytest.mark.parametrize(
+    "qin_arr", (0.5, [0.001, 0.37, 0.42, 0.67, 0.83, 0.99, 0.39, 0.49, 0.5])
+)
+@pytest.mark.parametrize("keepdims", (False, True))
+def test_quantiles_axis_none(str_method, qin_arr, keepdims):
+    eps = 1.0e-8
+    arr = np.ndarray(
+        shape=(2, 3, 4),
+        buffer=np.array(
+            [
+                1,
+                2,
+                2,
+                40,
+                1,
+                1,
+                2,
+                1,
+                0,
+                10,
+                3,
+                3,
+                40,
+                15,
+                3,
+                7,
+                5,
+                4,
+                7,
+                3,
+                5,
+                1,
+                0,
+                9,
+            ]
+        ),
+        dtype=int,
+    )
+
+    if cu.isscalar(qin_arr):
+        qs_arr = qin_arr
+    else:
+        qs_arr = np.array(qin_arr)
+
+    # cunumeric:
+    # print("cunumeric axis = %d:"%(axis))
+    q_out = cu.quantile(
+        arr,
+        qs_arr,
+        method=str_method,
+        keepdims=keepdims,
+    )
+    # print(q_out)
+
+    # np:
+    # print("numpy axis = %d:"%(axis))
+    np_q_out = np.quantile(
+        arr,
+        qs_arr,
+        method=str_method,
+        keepdims=keepdims,
+    )
+    # print(np_q_out)
+
+    assert q_out.shape == np_q_out.shape
+    assert q_out.dtype == np_q_out.dtype
+
+    assert allclose(np_q_out, q_out, atol=eps)
+
+
+@pytest.mark.parametrize(
+    "str_method",
+    (
+        "inverted_cdf",
+        "averaged_inverted_cdf",
+        "closest_observation",
+        "interpolated_inverted_cdf",
+        "hazen",
+        "weibull",
+        "linear",
+        "median_unbiased",
+        "normal_unbiased",
+        "lower",
+        "higher",
+        "midpoint",
+        "nearest",
+    ),
+)
+@pytest.mark.parametrize(
+    "qin_arr", (0.5, [0.001, 0.37, 0.42, 0.67, 0.83, 0.99, 0.39, 0.49, 0.5])
+)
+@pytest.mark.parametrize("axes", (None, 0))
+def test_random_inlined(str_method, qin_arr, axes):
+    eps = 1.0e-8
+    arr = cu.random.random((3, 4, 5))
+
+    q_out = cu.quantile(arr, qin_arr, method=str_method, axis=axes)
+    np_q_out = np.quantile(arr, qin_arr, method=str_method, axis=axes)
+
+    assert q_out.shape == np_q_out.shape
+    assert q_out.dtype == np_q_out.dtype
+
+    assert allclose(np_q_out, q_out, atol=eps)
+
+
```

Comment:
Let's add some one-off tests for some corner cases (no need to include them in all the combinations above):

* getting a quantile out of an array of shape `()` (i.e. if `arr` is a scalar)
* using an `out=` array of different type than what the input would suggest, e.g. input array is of type `np.float64` and output array is of type `np.float16`.
* behavior at the limits (0% and 100% quantile), something like: 
  ```
  np_a = np.arange(4)
  num_a = num.array(np_a)
  np_res = np.quantile(np_a, limit, method=method)
  num_res = num.quantile(num_a, limit, method=method)
  assert np_res == num_res
  ```

Reply:
All tests added. The 2nd one, on output conversion, is there but disabled (skipped) because `numpy` issue: https://github.com/numpy/numpy/issues/22766 must be addressed first in order to test (enable) this test.